benchmark/cajita/Cajita_InterpolationPerformance.cpp

```diff
@@ -0,0 +1,409 @@
+/****************************************************************************
+ * Copyright (c) 2018-2021 by the Cabana authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the Cabana library. Cabana is distributed under a   *
+ * BSD 3-clause license. For the licensing terms see the LICENSE file in    *
+ * the top-level directory.                                                 *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#include "../Cabana_BenchmarkUtils.hpp"
+
+#include <Cajita.hpp>
+
+#include <Kokkos_Core.hpp>
+
+#include <mpi.h>
+
+using namespace Cajita;
+
+//---------------------------------------------------------------------------//
+// Performance test.
+template <class Device>
+void performanceTest( std::ostream& stream, const std::string& test_prefix,
+                      std::vector<int> cells_per_dim,
+                      std::vector<int> particles_per_cell_dim )
+{
+    using exec_space = typename Device::execution_space;
+    using memory_space = typename Device::memory_space;
+
+    // Domain size setup
+    std::array<double, 3> global_low_corner = { 0.0, 0.0, 0.0 };
+    std::array<double, 3> global_high_corner = { 1.0, 1.0, 1.0 };
+    std::array<bool, 3> is_dim_periodic = { false, false, false };
+
+    // System size
+    int num_problem_size = cells_per_dim.size();
+
+    // Define the particle aosoas.
+    using member_types =
+        Cabana::MemberTypes<double[3][3], double[3], double[3], double>;
+    using aosoa_type = Cabana::AoSoA<member_types, Device>;
+
+    int num_particles_per_cell_dim = particles_per_cell_dim.size();
+
+    for ( int n = 0; n < num_problem_size; ++n )
+    {
+        // Create the global grid
+        double cell_size = 1.0 / cells_per_dim[n];
+        auto global_mesh = createUniformGlobalMesh(
+            global_low_corner, global_high_corner, cell_size );
+
+        Cajita::DimBlockPartitioner<3> partitioner;
+        auto global_grid = createGlobalGrid( MPI_COMM_WORLD, global_mesh,
+                                             is_dim_periodic, partitioner );
+
+        // Create a local grid and local mesh
+        int halo_width = 1;
+
+        auto local_grid = createLocalGrid( global_grid, halo_width );
+        auto local_mesh = createLocalMesh<exec_space>( *local_grid );
+        auto owned_cells = local_grid->indexSpace( Own(), Cell(), Local() );
+
+        int num_cells = owned_cells.size();
+
+        std::vector<aosoa_type> aosoas( num_particles_per_cell_dim );
```

Comment:
I think you may have already changed this locally, but let's not store all of the system sizes